From bf3511d59514aa7c58f0cef4cfeb8378ecc1df64 Mon Sep 17 00:00:00 2001
From: Steve Kondik <steve@cyngn.com>
Date: Sun, 31 Jul 2016 19:09:37 -0700
Subject: [PATCH 079/192] snap: Always turn touch-af-aec on

 * Most cameras do this by default, but on at least one device
   it starts with the option disabled. This stops AEC from
   updating during touch focus.

Change-Id: I4bdb6cdbad86f3798c0288d6a32b7e8062e297f3
---
 src/com/android/camera/PhotoModule.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index a5613ba72..394bda9fe 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -3225,6 +3225,12 @@ public class PhotoModule
         String optizoomOn = mActivity.getString(R.string
                 .pref_camera_advanced_feature_value_optizoom_on);
 
+        // Set Touch AF/AEC parameter.
+        if (CameraUtil.isSupported(mParameters.TOUCH_AF_AEC_ON,
+                mParameters.getSupportedTouchAfAec())) {
+            mParameters.setTouchAfAec(mParameters.TOUCH_AF_AEC_ON);
+        }
+
         // Set Picture Format
         // Picture Formats specified in UI should be consistent with
         // PIXEL_FORMAT_JPEG and PIXEL_FORMAT_RAW constants
-- 
2.17.1

